Test failures due to Java deprecation warnings

I'm trying to run the test suite following [wiki/INSTALL](https://github.com/dafny-lang/dafny/wiki/INSTALL) and I see the following failures, stable across reruns:

```
Failed Tests (15):
  Dafny :: comp/Arrays.dfy
  Dafny :: comp/AutoInit.dfy
  Dafny :: comp/Calls.dfy
  Dafny :: comp/Class.dfy
  Dafny :: comp/Collections.dfy
  Dafny :: comp/Extern.dfy
  Dafny :: comp/MainMethod.dfy
  Dafny :: comp/MoreAutoInit.dfy
  Dafny :: comp/NativeNumbers.dfy
  Dafny :: comp/Numbers.dfy
  Dafny :: comp/StaticMembersOfGenericTypes.dfy
  Dafny :: comp/TypeDescriptors.dfy
  Dafny :: dafny0/NonZeroInitializationCompile.dfy
  Dafny :: git-issues/git-issue-403.dfy
  Dafny :: server/counterexample_none.transcript


Testing Time: 1556.87s
  Unsupported      :    1
  Passed           : 1046
  Expectedly Failed:    3
  Failed           :   15
```

Running `lit` with `--show-all` shows that they are all caused by unexpected Java deprecation warnings, for example:
```
+/Users/fpoli/Documents/projects/dafny/Test/comp/Calls-java/FunctionValues_Compile/NT.java:12: warning: [removal] Byte(byte) in Byte has been deprecated and marked for removal
+    for (java.math.BigInteger j = lo; j.compareTo(hi) < 0; j.add(java.math.BigInteger.ONE)) { arr.add(new Byte(j.byteValue())); }
+                                                                                                      ^
+1 warning
```

Is there a recommended JDK version that should be used to run the test suite? I tried Amazon Corretto version 8, 11 and 16. With the first two the 15 tests that I reported above fail; with version 16 `make runtime` fails with an exception.

--- a/Source/Dafny/Compiler-java.cs
+++ b/Source/Dafny/Compiler-java.cs
@@ -3385,7 +3385,7 @@ namespace Microsoft.Dafny{
         var numberval = "intValue()";
         if (nativeType == "Byte" || nativeType == "Short")
           numberval = $"{nativeType.ToLower()}Value()";
-        wEnum.WriteLine($"for (java.math.BigInteger j = lo; j.compareTo(hi) < 0; j.add(java.math.BigInteger.ONE)) {{ arr.add(new {nativeType}(j.{numberval})); }}");
+        wEnum.WriteLine($"for (java.math.BigInteger j = lo; j.compareTo(hi) < 0; j.add(java.math.BigInteger.ONE)) {{ arr.add({nativeType}.valueOf(j.{numberval})); }}");
         wEnum.WriteLine("return arr;");
       }
       if (nt.WitnessKind == SubsetTypeDecl.WKind.Compiled) {

